### ISSUE TITLE
`valid-v-for`: exclude lodash's `uniqueID()`

### ISSUE THREAD

**Issue by MetRonnie** [vuejs/eslint-plugin-vue#2246]
**What rule do you want to change?**

`vue/valid-v-for`

**Does this change cause the rule to produce more or fewer warnings?**

Fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**

Do not warn if we v-bind `:key="uniqueId()"`

**Please provide some example code that this change will affect:**

We are iterating over an array containing some identical items (so we cannot use any property of the items as the key as that would risk Vue losing track of the items)
```vue
let data = [A, A, B]
<div v-for="item in data" :key="uniqueId()">
```

**What does the rule currently do for this code?**#

Produces this warning:
> Expected 'v-bind:key' directive to use the variables which are defined by the 'v-for' directive

**What will the rule do after it's changed?**

Not produce a warning for the value `uniqueId()`.

**Additional context**

https://stackoverflow.com/a/75175749/3217306

**Comment by FloEdelmann** [comment]
I think the rule is working as intended, and using a random unique ID is not really the intended use case of the `key` directive.

You can still disable the rule, either globally in the ESLint config file, or locally just for one component with the `<!-- eslint-disable-next-line vue/valid-v-for -->` comment.

**Comment by FloEdelmann** [comment]
It depends whether the list rendering should change when the component state or props etc. changed. If it shouldn't (because the array is static), then you could simply use the array index as the key. If it should, then the whole list requires a re-rendering.

Using `uniqueId()` as each item's key will always force a re-rendering, regardless whether it is necessary or not. So I think it would be better to either add a `key` the parent element (forcing a re-rendering of all child elements) with something that depends on the component state (plus using the array indices as item keys), or use e.g. `` `${componentStateVariable}-${arrayIndex}` `` as the item keys.

See also:

* https://vuejs.org/api/built-in-special-attributes.html#key
* https://vuejs.org/guide/essentials/list.html#maintaining-state-with-key
